File: market-data-public/exchange-trades/exchange-trades.py
import requests
import time

def fetch_exchange_trades(market_code, limit=100):
    # Calculate startTime as 24 hours ago
    start_time = int(time.time() * 1000) - 24 * 60 * 60 * 1000  # 24 hours in milliseconds

    # Calculate endTime as the current time
    end_time = int(time.time() * 1000)

    url = f'https://api.ox.fun/v3/exchange-trades?marketCode={market_code}&limit={limit}&startTime={start_time}&endTime={end_time}'

    try:
        response = requests.get(url)
        response.raise_for_status()
        print('Exchange Trades Data:', response.json())
    except requests.exceptions.RequestException as error:
        print('Error fetching exchange trades data:', error)

# Example usage
fetch_exchange_trades('BTC-USD-SWAP-LIN', 100)

File: market-data-public/exchange-trades/exchange-trades.js
const axios = require('axios');

async function fetchExchangeTrades(marketCode, limit = 100) {
  // Calculate startTime as 24 hours ago
  const startTime = Date.now() - 24 * 60 * 60 * 1000; // 24 hours in milliseconds

  // Calculate endTime as the current time
  const endTime = Date.now();

  const url = `https://api.ox.fun/v3/exchange-trades?marketCode=${marketCode}&limit=${limit}&startTime=${startTime}&endTime=${endTime}`;

  try {
    const response = await axios.get(url);
    console.log('Exchange Trades Data:', response.data);
  } catch (error) {
    console.error('Error fetching exchange trades data:', error.response ? error.response.data : error.message);
  }
}

// Example usage
fetchExchangeTrades('BTC-USD-SWAP-LIN', 100);